File: Deepseek-Coder-33b-Instruct/stage2/acc_wait_any.cpp
#include <openacc.h>
#include <iostream>

#ifndef T1
//T1:acc wait any,V:2.7-3.3
int test1(){
    int err = 0;
    int count = 10;
    int wait_arg[count];

    // Initialize the async arguments
    for (int i = 0; i < count; ++i) {
        wait_arg[i] = acc_async_sync;
    }

    // Wait for any async argument to complete
    int index = acc_wait_any(count, wait_arg);

    // Check if the returned index is valid
    if (index < 0 || index >= count) {
        err = 1;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/acc_wait_any.f90
program test_acc_wait_any
  use openacc
  implicit none

  integer, parameter :: n = 1000
  integer :: i, async1, async2, wait_arg(2), count, dev_num
  real :: a(n), b(n), c(n)

  ! Initialize data
  do i = 1, n
    a(i) = i
    b(i) = i * 2
  end do

  ! Start asynchronous tasks
  !$acc parallel async(1) copyin(a) copyout(c)
  async1 = acc_get_async_info()
  c(:) = a(:)
  !$acc end parallel

  !$acc parallel async(2) copyin(b) copyout(c)
  async2 = acc_get_async_info()
  c(:) = c(:) + b(:)
  !$acc end parallel

  ! Wait for any of the tasks to complete
  wait_arg(1) = async1
  wait_arg(2) = async2
  count = 2
  dev_num = acc_get_device_num(acc_device_nvidia)
  call acc_wait_any_device(count, wait_arg, dev_num)

  ! Check the result
  do i = 1, n
    if (c(i) .ne. 3 * i) then
      print *, "Test failed"
      stop 1
    end if
  end do

  print *, "Test passed"
end program test_acc_wait_any